### DISCUSSION TITLE
Welcome to github-actions-issue-to-work-item Discussions!

### DISCUSSION THREAD

**Discussion by danhellem** [danhellem/github-actions-issue-to-work-item#62] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by sandeepmalu** [comment]
How to use generic user Github and Azure PAT to sync issue between Github and Azure board for organization Github account?

I am trying this github actions. It can access github.actor Github PAT but fails to use github.actor Azure PAT. Though it works with user specified Azure PAT

**************************************************************************************************
name: Sync issue to Azure DevOps work item

on:
  issues:
    types:
      [opened, edited, deleted, closed, reopened, labeled, unlabeled, assigned]

jobs:
  alert:
    runs-on: ubuntu-latest
  
    steps:
      - name: echo github PAT
        env:
          github_username_token: ${{ format('secrets.{0}_GITHUB_PERSONAL_ACCESS_TOKEN',github.actor)  }}
        run: |
          echo ${{ env.github_username_token }}
      - name: echo azure PAT
        env:
          azure_username_token: ${{ format('secrets.{0}_AZURE_PERSONAL_ACCESS_TOKEN',github.actor)  }}
        run: |
          echo ${{ env.azure_username_token }}
      - uses: danhellem/github-actions-issue-to-work-item@master
        env:
          ado_token: "${{ env.azure_username_token }}"
          github_token: "${{ env.github_username_token }}"
          ado_organization: "Software-Engineering"
          ado_project: "All Software Development"
          ado_wit: "Issue"
          ado_new_state: "New"
          ado_active_state: "Active"
          ado_close_state: "Closed"
          ado_bypassrules: true

**************************************************************************************************************************
I mean "env.azure_username_token" doesn't work

**Reply by danhellem** [reply]
What do you mean by "it does not work"? Are you getting errors?

**Reply by sandeepmalu** [reply]
Yes, It fails when we use "env.azure_username_token" variable but if we use "SPECIFIC_USER_ADO_PERSONAL_ACCESS_TOKEN" mentioned in Github secrets it works. Actually, we have multiple users on Github repo and we want to use ADO_PAT of the User which creates an issue rather than One ADO_PAT for all. That's why we are using github.actor to make it generic and it doesn't work.

**Reply by danhellem** [reply]
@sandeepmalu looks like you created issue #74 for this. Will try and get it done over the next couple of weeks for us.

**Reply by sandeepmalu** [reply]
@danhellem Thank you

**Reply by danhellem** [reply]
@sandeepmalu go ahead and get latest and see if that helps

**Comment by omharimca** [comment]
When I tried same approach which is given here for "github-actions-issue-to-work-item" to integrate the ADO to git issue. 
I'm getting below error msg -
_**danhellem/github-actions-issue-to-work-item@master is not allowed to be used in ansible-playbooks/devops.proxy_server. Actions in this workflow must be: within a repository that belongs to your Enterprise account, created by GitHub, verified in the GitHub Marketplace or match the following: DeLaGuardo/setup-clojure@3.2, SonarSource/sonarqube-quality-gate-action@v1**_

Can some one plz help to resolve this issue step by step.

**Reply by danhellem** [reply]
It looks like you have a policy that does not allow outside actions to be applied to the account. Try using the version number instead of @master

danhellem/github-actions-issue-to-work-item@v2.

**Comment by vman92** [comment]
Good Afternoon, I am a scrum master on a infrastructure team using GitHub. GitHub doesn't have near the metrics our management wants. So, I was looking at Zenhub as an option. That didn't go anywhere. Then I built a dashboard using Codex. It is ok. Now we are at a place where our product owners are at least willing to use AzDO (_cloud_) even if the engineers can still stay in GitHub. So, this connection is very interesting to me. We have implemented the GitHub Project overlay and manage estimates, iterations, etc. in there.  We use a board that surfaces multiple repos/

In the demo video, I see an issue created in GitHub and it is automatically created in AzDO.  (Nice)
I see what looks to be labels populating in AzDO, issue or PBI state (ie. New) and possibly iteration?

Can I ask; does this work with syncing data from multiple GitHub repos to one AzDO board?
Does this sync up comments in GitHub issues with the associated AzDO story?
Is issue state synced with the AzDO board?
Is GitHub Project data synced?

Thanks
